src/dot_parser/mod.rs

```diff
@@ -0,0 +1,299 @@
+                                    dst_idx,
+                                    edge_attrs_obj.clone(),
+                                );
+                            } else {
+                                undi_graph.as_mut().unwrap().graph.add_edge(
+                                    src_idx,
+                                    dst_idx,
+                                    edge_attrs_obj.clone(),
+                                );
+                            }
+                        }
+                    }
+
+                    Rule::attr_stmt => {
+                        let mut it = stmt.into_inner();
+                        if let Some(target_pair) = it.next() {
+                            let target = target_pair.as_str();
+                            for rest in it {
+                                if rest.as_rule() == Rule::attr_list {
+                                    let map = parse_attr_list_to_map(rest);
+                                    match target {
+                                        "graph" => {
+                                            for (k, v) in map {
+                                                graph_attrs.set_item(k, v)?;
+                                            }
+                                            if let Some(dg) = di_graph.as_mut() {
+                                                dg.attrs = graph_attrs.clone().into();
+                                            }
+                                            if let Some(ug) = undi_graph.as_mut() {
+                                                ug.attrs = graph_attrs.clone().into();
+                                            }
+                                        }
+                                        "node" => {
+                                            for (k, v) in map {
+                                                default_node_attrs.insert(k, v);
+                                            }
+                                        }
+                                        "edge" => {
+                                            for (k, v) in map {
+                                                default_edge_attrs.insert(k, v);
+                                            }
+                                        }
+                                        _ => {}
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    Rule::assignment => {
+                        let mut parts = stmt.into_inner();
+                        let key = parts.next().map(|p| p.as_str()).unwrap_or("");
+                        let val = parts.next().map(|p| p.as_str()).unwrap_or("");
+                        graph_attrs.set_item(key, val)?;
+                    }
+
+                    Rule::subgraph => {
+                        // TODO: subgraph handling
```

Comment:
It's ok to return an error for subgraphs now. `dot` might be too expresive. We just want to make sure the output from a rustworkx entry can be re-imported as the original goal.

Once we have that, we can support more advanced features